libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -502,7 +534,39 @@ bool ExpressionCompiler::visit(BinaryOperation const& _binaryOperation)
 	CompilerContext::LocationSetter locationSetter(m_context, _binaryOperation);
 	Expression const& leftExpression = _binaryOperation.leftExpression();
 	Expression const& rightExpression = _binaryOperation.rightExpression();
-	solAssert(!!_binaryOperation.annotation().commonType, "");
+	FunctionDefinition const* function = *_binaryOperation.annotation().userDefinedFunction;
+	if (function)
+	{
+		solAssert(function->isFree() || function->libraryFunction());
+
+		FunctionType const* functionType = _binaryOperation.userDefinedFunctionType();
+		solAssert(functionType);
+		functionType = dynamic_cast<FunctionType const&>(*functionType).withBoundFirstArgument();
```

Comment:
Same here. Even though we reuse the `using` syntax, operators are not really *bound* in any way, let alone in one of two operand arguments...